### ISSUE TITLE
Run select and iocp reactor tests on Appveyor

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8677]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#8677|
|Type|enhancement|
|Created|2016-07-25 16:28:40Z|
|Branch|https://github.com/twisted/twisted/tree/8677-rodrigc-appveyor-iocp|

to match the buildbot builder

<details><summary>Searchable metadata</summary>

```
trac-id__8677 8677
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__None None
branch__8677_rodrigc_appveyor_iocp 8677-rodrigc-appveyor-iocp
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1469464120768807 1469464120768807
changetime__1474198049187875 1474198049187875
version__None None
owner__Craig_Rodrigues__rodrigc_____ Craig Rodrigues <rodrigc@...>

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)|@adiroiban set owner to @adiroiban<br>@adiroiban set status to `assigned`|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_2"></a>|@adiroiban removed owner<br>@adiroiban set status to `new`|
|-|-|

PR at https://github.com/twisted/twisted/pull/412

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

OK. So... enabling build extensions should come together with running iocpreactor tests as in this way we can check that the extension was compiled and installed.

I have enabled extentions on windows (using a support script for tox) and ignored iocp reactor tests. They can be fixed in [#8679](https://github.com/twisted/twisted/issues/8679)

Please check that this is a good start for having iocpreactor tests on Windows.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @adiroiban|
|-|-|

Thanks for doing this, Adi.  Running these builders on appveyor would be great.

Craig put some comments on the github branch which I pretty much agree with; could you address those?

I'd like to see what the config changes look like after that, so please address and resubmit.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_5"></a>|@adiroiban removed owner|
|-|-|

Craig, Glyph, many thanks for the review.

I have addressed the comments.

I have removed the python3.5 definitions as they were commented and not used. I have also removed other command which were commented, in an attempt to clean the script.

Is there a ticket for enabling python3.5 on Appeyor? I can look at enabling that build, and put it in allow_failures as a start.

CMD_IN_ENV wrapping bin\admin\run_with_env.cmd wrapping python.bat which wraps python.exe which then calls tox ... is one ugly chain


Thanks again for the review!

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_6"></a>|@adiroiban commented|
|-|-|

Ah... I forgot.

Glyph, do you know why do we have ./admin and ./bin/admin folders ?
Once I resolve this mister I can add a README file in each folder to document their purpose.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_7"></a>|@glyph commented|
|-|-|

Replying to [adiroiban](#note_6):
> Ah... I forgot.
> 
> Glyph, do you know why do we have ./admin and ./bin/admin folders ?
> Once I resolve this mister I can add a README file in each folder to document their purpose.
> 
> Thanks!

I have no idea.  When I created the directory structure initially, there was `bin/` for stuff that got distributed to users, and `admin/` for stuff that we used on development of the project itself.  As far as I'm concerned, `bin/admin/` should be removed.  (And there's no compatibility concern here, since that stuff isn't even distributed, let alone having an API surface.)

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_8"></a>|@adiroiban commented|
|-|-|

OK. I will look at cleaning the bin/admin as each time I add an admin script I don't know if it should go in admin or bin/admin

[#8729](https://github.com/twisted/twisted/issues/8729) is the ticket for enabling python3 on Appveyor

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_9"></a>|@rodrigc commented|
|-|-|

I have a much simpler patch here:

https://github.com/twisted/twisted/pull/485

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_10"></a>|@rodrigc set owner to @rodrigc<br>@rodrigc set status to `closed`|
|-|-|

In changeset defe32ce6ed4b89beb8bff10c465782019813d9b
```
#!CommitTicketReference repository="" revision="defe32ce6ed4b89beb8bff10c465782019813d9b"
Merge pull request #485 from twisted/8677-rodrigc-appveyor-iocp

Author: rodrigc
Reviewer: opalmer
Fixes: #8677

Update appveyor config to test select and iocp reactors
```

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_11"></a>|@hawkowl commented|
|-|-|

[mass edit] Removing review from closed tickets.
